services/channel.go

```diff
@@ -0,0 +1,45 @@
+package services
+
+import (
+	"github.com/streampets/backend/models"
+)
+
+type Event struct {
+	Event   string
+	Message string
+}
+
+type EventStream chan Event
+
+type TwitchRepo interface {
+	GetUsername(channelID models.UserID) (string, error)
+}
+
+type ChannelService struct {
+	twitchRepo TwitchRepo
+	streams    map[string]EventStream
+}
+
+func NewChannelService(twitchRepo TwitchRepo) *ChannelService {
+	return &ChannelService{
+		twitchRepo: twitchRepo,
+		streams:    make(map[string]EventStream),
+	}
+}
+
+func (s *ChannelService) GetEventStream(channelID models.UserID) (EventStream, error) {
+	channelName, err := s.twitchRepo.GetUsername(channelID)
+	if err != nil {
+		return nil, err
+	}
+
+	stream, ok := s.streams[channelName]
+	if ok {
```

Comment:
Go question - can channels be closed? Does this need to check whether the channel is closed prior to returning it? Maybe creating another if it's closed?

Reply:
Channels can be closed, but there isn't an obvious way to check if they are
Since we're managing the channels ourselves within this struct, we know that we're not closing them, so they should never be closed